### PR TITLE
Add implementation template images

### PR DESCRIPTION
**Pull request by ValeriiMalkov**
Add implementation template images
on the HMI for next RPC:

- SetGlobalProperties
- AddCommand
- Show
- ShowConstantTBT
- SendLocation
- UpdateTurnList
- Alert
- ScrollableMessage
- AlertManeuver
- CreateInteractionChoiceSet

### CODE REVIEW

File: app/model/sdl/Abstract/Model.js [hunk 1]
```diff
@@ -881,10 +881,10 @@ SDL.SDLModel = Em.Object.extend({
 
       var img = new Image();
       img.onload = function() {
-
+        var model=SDL.SDLController.getApplicationModel(message.appID);
         // code to set the src on success
-        SDL.SDLController.getApplicationModel(message.appID).
-            set('appIcon', message.syncFileName.value);
+        model.set('appIcon', message.syncFileName.value);
+        model.set('isTemplateIcon',false);
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/Model.js]: @ValeriiMalkov this parameter should depend on isTemplate parameter in request:
```
model.set('isTemplateIcon', message.syncFileName.isTemplate === true);
```

File: app/model/sdl/MediaModel.js [hunk 2]
```diff
@@ -341,14 +354,24 @@ SDL.SDLMediaModel = SDL.ABSAppModel.extend({
...
+          if(isPng){
       if (params.graphic.value != '') {
         this.appInfo.set('trackIcon', params.graphic.value);
       } else {
         this.appInfo.set('trackIcon', 'images/sdl/audio_icon.jpg');
       }
+      if(params.graphic.isTemplate!=null){
+        this.set('isTemplate','DYNAMIC'==params.graphic.imageType ? params.graphic.isTemplate : false);
+      }
+      else {this.set('isTemplate',false);}
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/MediaModel.js]: @ValeriiMalkov this if-else could be replaced with one line:
```
this.set('isTemplate', 'DYNAMIC' == params.graphic.imageType && params.graphic.isTemplate === true);
```

File: app/view/home/controlButtons.js [hunk 3]
```diff
@@ -53,8 +53,64 @@ SDL.ControlButtons = Em.ContainerView.create({
...
+      SDL.InfoView.leftMenu.items.appsButton.setMode(this.selection);
+      SDL.InfoView.leftMenu.items.calendarButton.setMode(this.selection);
+      SDL.InfoView.leftMenu.items.goToCD.setMode(this.selection);
+      SDL.InfoView.leftMenu.items.travelLinkButton.setMode(this.selection);
+      SDL.InfoView.leftMenu.items.sdlButton.setMode(this.selection);
+      SDL.TurnByTurnView.nextTurnIconImage.setMode(this.selection);
+      SDL.TurnByTurnView.turnIconImage.setMode(this.selection);
+      SDL.InteractionChoicesView.set('imageMode',this.selection);
+      SDL.InteractionChoicesView.updateIcons();
+      if(null!=SDL.SDLController.model){
```
Comment (AKalinich-Luxoft) [inline app/view/home/controlButtons.js]: @ValeriiMalkov just:
```
if(SDL.SDLController.model){
```

File: app/view/home/controlButtons.js [hunk 4]
```diff
@@ -53,8 +53,64 @@ SDL.ControlButtons = Em.ContainerView.create({
...
+      SDL.SDLController.model.setMode(this.selection);
+      }
+      if (SDL.SDLController.model) {
```
Comment (AKalinich-Luxoft) [inline app/view/home/controlButtons.js]: @ValeriiMalkov the same condition second time. Please merge this if with previous

File: app/view/sdl/shared/scrollableMessage.js [hunk 5]
```diff
@@ -63,10 +64,14 @@ SDL.ScrollableMessage = SDL.SDLAbstractView.create(
       this.set('active', false);
       this.softButtons.set('page', 0);
       this.timeout = null;
+      if(!this.warning){
       SDL.SDLController.scrollableMessageResponse(
         ABORTED ? SDL.SDLModel.data.resultCode['ABORTED'] :
           SDL.SDLModel.data.resultCode.SUCCESS, this.messageRequestId
-      );
+      );}else{ SDL.SDLController.scrollableMessageResponse(
```
Comment (AKalinich-Luxoft) [inline app/view/sdl/shared/scrollableMessage.js]: @ValeriiMalkov fix this formatting

File: css/general.css [hunk 6]
```diff
@@ -41,6 +41,48 @@ div {
...
+#tbtTurnListView .list-item.high-lighted-mode .ico-overlay
+{
+    background:url(../images/template/imagetype-pattern-highlighted.png);
+    top: 0px;
+    left: 0px;
+    position: absolute;
+    background-position: -6px -11px;
+    background-repeat: round;
+    opacity: 0.4;
+    /* width: 100px;
```
Comment (AKalinich-Luxoft) [inline css/general.css]: @ValeriiMalkov remove commented attributes

File: ffw/NavigationRPC.js [hunk 7]
```diff
@@ -490,7 +557,7 @@ FFW.Navigation = FFW.RPCObserver.create(
         return;
       }
       Em.Logger.log('FFW.UI.' + method + 'Response');
-      if (resultCode === SDL.SDLModel.data.resultCode.SUCCESS) {
+      if (resultCode === SDL.SDLModel.data.resultCode.SUCCESS ||SDL.SDLModel.data.resultCode.WARNINGS ) {
```
Comment (AKalinich-Luxoft) [inline ffw/NavigationRPC.js]: @ValeriiMalkov please specify explicitly:
```
(resultCode == SDL.SDLModel.data.resultCode.SUCCESS || resultCode == SDL.SDLModel.data.resultCode.WARNINGS )
```

File: app/controller/sdl/Abstract/Controller.js [hunk 8]
```diff
@@ -50,6 +50,7 @@ SDL.SDLController = Em.Object.extend(
      * Active application model binding type {SDLAppModel}
      */
     model: null,
+    isWarning:'',
```
Comment (AKalinich-Luxoft) [inline app/controller/sdl/Abstract/Controller.js]: @ValeriiMalkov please consider using bool value type here

File: app/view/home/controlButtons.js [hunk 9]
```diff
@@ -53,8 +53,64 @@ SDL.ControlButtons = Em.ContainerView.create({
...
+    default: return '';
+  }
+},
+/**
+ * HMI element Select with list of supported image mode
+ */
+  imageMode:Em.Select.extend({
+    elementId: 'imageMode',
+    classNames: 'imageModeSelect',
+    contentBinding: 'SDL.SDLModel.data.imageModeList',
```
Comment (AKalinich-Luxoft) [inline app/view/home/controlButtons.js]: @ValeriiMalkov please add:
```
selection: 'Highlighted mode',
```

File: app/model/sdl/Abstract/Model.js (hunk 1 above)
Comment (ValeriiMalkov) [reply app/model/sdl/Abstract/Model.js]: @AKalinich-Luxoft it was fixed at the commit 86d275da375fb72f2a907b76262b1133a7e5fe3c

File: app/model/sdl/MediaModel.js (hunk 2 above)
Comment (ValeriiMalkov) [reply app/model/sdl/MediaModel.js]: @AKalinich-Luxoft  it was fixed at the commit 86d275d

File: app/view/home/controlButtons.js (hunk 3 above)
Comment (ValeriiMalkov) [reply app/view/home/controlButtons.js]: @AKalinich-Luxoft it was fixed at the commit 86d275d

File: app/view/home/controlButtons.js (hunk 4 above)
Comment (ValeriiMalkov) [reply app/view/home/controlButtons.js]: @AKalinich-Luxoft it was fixed at the commit 86d275d

File: app/view/sdl/shared/scrollableMessage.js (hunk 5 above)
Comment (ValeriiMalkov) [reply app/view/sdl/shared/scrollableMessage.js]: @AKalinich-Luxoft it was fixed at the commit 86d275d

File: css/general.css (hunk 6 above)
Comment (ValeriiMalkov) [reply css/general.css]: @AKalinich-Luxoft it was fixed at the commit 86d275d

File: ffw/NavigationRPC.js (hunk 7 above)
Comment (ValeriiMalkov) [reply ffw/NavigationRPC.js]: @AKalinich-Luxoft it was fixed at the commit 86d275d

File: app/controller/sdl/Abstract/Controller.js (hunk 8 above)
Comment (ValeriiMalkov) [reply app/controller/sdl/Abstract/Controller.js]: @AKalinich-Luxoft it was fixed at the commit 86d275d

**Pull request closed without merging**